mate-panel/panel-multiscreen.c

```diff
@@ -483,12 +490,23 @@ panel_multiscreen_screens (void)
 	return screens;
 }
 
+inline int
+_get_screen_number (GdkScreen *screen)
+{
+#ifdef HAVE_X11
+	if (GDK_IS_X11_DISPLAY (gdk_screen_get_display (screen)))
+		return gdk_x11_screen_get_screen_number (screen);
+#endif
+
+	return gdk_screen_get_number (screen);
```

Comment:
`gdk_screen_get_number` is deprecated

There is always one screen, I think `gdk_x11_screen_get_screen_number`  and `gdk_screen_get_number` can be replaced safely by the value 0

Reply:
It might work, but you can not do that safely...

On X11 you can have one display with n screens (separate x screens with NVIDIA is something that I have used in past). For example if you have two x screens (:0.0 and :0.1) you could start panel with `DISPLAY=:0.1 mate-panel`. It would start panel on display 0 and screen 1. In gtk you would still get one display and one screen, but screen number will be 1 not 0.

https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gdk/x11/gdkscreen-x11.c#L906
https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gdk/x11/gdkdisplay-x11.c#L1602
https://tronche.com/gui/x/xlib/display/opening.html

Of course these days mostly it is one display, one screen with one or more monitors.

Reply:
I'll leave the X code as-is, and change the non-X code to just return 0.